core/utils.go

```diff
@@ -1 +1,80 @@
 package core
+
+import (
+	"fmt"
+	"github.com/refractionPOINT/go-limacharlie/limacharlie"
+	"path"
+	"strings"
+	"sync"
+)
+
+var (
+	secretCache = make(map[string]string)
+	cacheMutex  sync.RWMutex
+)
+
+func usingSecretValue(key string, org *limacharlie.Organization, fn func(val string) error) error {
+	var err error
+	var apiKey string
+	var exists bool
+	if strings.Contains(key, "hive://") {
+		secretName := path.Base(key)
+		// Try to get secret from cache
+		apiKey, exists = getSecretFromCache(secretName)
+		if !exists {
+			apiKey, err = getSecretFromHive(secretName, org)
+			if err != nil {
+				return err
+			}
+
+			// ensure to update local cache
+			setSecretCache(secretName, apiKey)
+		}
+
+		const maxRetries = 2
+		for i := 0; i < maxRetries; i++ {
+			err = fn(apiKey)
```

Comment:
Hmm, the resolution of the Hive secret needs to be done in the retry loop so that if we get an error using the secret, we attempt to fetch an updated value.

Reply:
opps logic added.

Reply:
Instead of doing it twice (once outside the loop and once after the tx in the loop), you can just do it once at the beginning of the loop. Just a bit cleaner.